src/dct/goals/DamageGoal.lua

```diff
@@ -70,21 +57,22 @@ end
 
 function DamageGoal:_afterspawn()
 	local obj, getlife = getobject(self.objtype, self.name)
-	if obj == nil then
-		Logger:error("_afterspawn() - object '%s' doesn't exist", self.name)
+	if obj == nil or not Object.isExist(obj) and not Group.isExist(obj) then
```

Comment:
Important note: `Object.isExist` covers `Scenery.isExist` and `Unit.isExist` but not `Group.isExist`, so we need to call both here, otherwise group damage goals will be completed on spawn

Reply:
I need to test this with `StaticObject` as well...

Reply:
I suspect `StaticObject` is covered by `Object.isExist`. It appears only `Group` is a special case in the class hierarchy, always good to check though. Please update when you have verified `StaticObject` is tested.

Reply:
I just verified it in-game and it's correctly covering `StaticObject` as well.